WIP Native Events

kuceb opened the pull request:
fix #311 

- typing into datetime-local input type
fix #2613


- input events for `{enter}`: 
fix #3405
- sending `code` property on keyboard event
fix #3722
- [ ] follow w3 spec for number inputs
fix #4767
### Tasks
- [x] waiting on #4103 use webpack internally
- [x] waiting on #2956 simulated click fixes
- [ ] pull in changes from develop + type_spec.coffee
- [ ] factor out synthetic type changes

jennifer-shehane commented on packages/driver/src/cy/actionability.coffee:
```diff
@@ -186,6 +186,7 @@ ensureElIsNotCovered = (cy, win, $el, fromViewport, options, log, onScroll) ->
   return $elAtCoords
 
 getCoordinatesForEl = (cy, $el, options) ->
+  debugger
```
Y'all need some linters running on precommit hooks, debugger should never been committed. 👀

Bkucera replied:
its a WIP, so i use `--no-verify`

Bkucera replied:
I don't think we should bother reviewing WIP PR's, isn't that the point of WIP? :smile:

brian-mann replied:
Agree with @Bkucera 	if we indicate a WIP on a PR it should be ignored and not be reviewed.

kuceb committed:
click mostly working

kuceb committed:
remove babelify code [skip ci]

kuceb committed:
add keyhole to native clicks

kuceb committed:
temp 11/09/18

kuceb committed:
Merge remote-tracking branch 'origin/develop' into issue-311

kuceb committed:
decaf

kuceb committed:
decaffeinate: Rename files

decaffeinate: Rename keyboard.coffee from .coffee to .js

decaffeinate: Rename click.coffee from .coffee to .js

decaffeinate: Rename type.coffee from .coffee to .js

kuceb committed:
decaffeinate: convert files

decaffeinate: Convert type.coffee to JS

decaffeinate: Run post-processing cleanups on type.coffee

decaffeinate: Convert keyboard.coffee to JS

decaffeinate: Run post-processing cleanups on keyboard.coffee

decaffeinate: Convert click.coffee to JS

decaffeinate: Run post-processing cleanups on click.coffee

kuceb committed:
Merge branch 'decaf-issue-311' into temp-311-done

kuceb committed:
decaffeinate: Rename selection.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert selection.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on selection.coffee

kuceb committed:
Merge branch 'decaf-issue-311' into issue-311

kuceb committed:
delete selection.js

kuceb committed:
decaffeinate: Rename selection.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert selection.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on selection.coffee

kuceb committed:
temp

kuceb committed:
temp 11/28/18

kuceb committed:
native click and type [skip ci]

kuceb committed:
fix programmatic blur events, allow typing into currently focused, fix getHostContenteditable

kuceb committed:
intercept .blur

kuceb committed:
decaffeinate: Rename click.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert click.coffee to JS

kuceb committed:
decaffeinate: Convert click.coffee to JS

decaffeinate: Convert mouse.coffee to JS

decaffeinate: Convert keyboard.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on click.coffee

decaffeinate: Run post-processing cleanups on mouse.coffee

decaffeinate: Run post-processing cleanups on keyboard.coffee

kuceb committed:
decaffeinate: Run post-processing cleanups on click.coffee

kuceb committed:
decaffeinate: Rename mouse.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert mouse.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on mouse.coffee

kuceb committed:
decaffeinate: Rename keyboard.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert keyboard.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on keyboard.coffee

kuceb committed:
reference issues in tests

kuceb committed:
update click to send all necessary events

kuceb committed:
add pointerevents, handle preventDefault

kuceb committed:
decaffeinate: rename click, keyboard, mouse

kuceb committed:
decaffeinate: Rename click_spec.coffee from .coffee to .js

kuceb committed:
decaffeinate: Convert click_spec.coffee to JS

kuceb committed:
decaffeinate: Run post-processing cleanups on click_spec.coffee

kuceb committed:
add tests, escapse strings for console

kuceb committed:
add comment about tests in logger.js

kuceb committed:
fix undefined prop bug

kuceb committed:
fix broken test

kuceb committed:
move keyboard, mouse under cy/

kuceb committed:
recalculate coords after "nudge" scroll
algorithm

kuceb committed:
fix corrupting test in click_spec

kuceb committed:
finish moving keyboard/mouse to cy/

kuceb committed:
remove .only from type_spec

kuceb committed:
fix up click_spec, fix edge cases

kuceb committed:
add "be.focused" to chai_jquery

kuceb committed:
add comment explaining the mess

kuceb committed:
add type definitions

kuceb committed:
Merge remote-tracking branch 'origin/issue-97-focus-assertion' into issue-2957

kuceb committed:
refactor assertion to use jquery assertion

jennifer-shehane committed:
remove trailing whitespace

kuceb committed:
add test for multiple elements, update typedefs

kuceb committed:
fix failing tests: not.be.visible -> not.exist

kuceb committed:
allow should(not.be.visible) for failed selectors

kuceb committed:
remove unrelated visibility changes

kuceb committed:
Merge remote-tracking branch 'origin/develop' into issue-97-focus-assertion

kuceb committed:
extract only focus assertion

kuceb committed:
Revert "Merge remote-tracking branch 'origin/develop' into issue-97-focus-assertion"

This reverts commit b401c32a6120cf67d01c2c36e5d605e99ee5272e, reversing
changes made to ab14758d3192433373b7f7588e98fe87fbed4e6c.

kuceb committed:
Merge remote-tracking branch 'origin/issue-97-focus-assertion' into issue-2957

kuceb committed:
temp 02/01/19

kuceb committed:
polish for review

kuceb committed:
play nice with consoleProps for cy.type

kuceb committed:
add dblclick + tests

kuceb committed:
rightClick, focus test changes

kuceb committed:
Merge remote-tracking branch 'origin/click-command-focus' into issue-2957

kuceb committed:
Revert "rightClick, focus test changes"

This reverts commit cdb464ed6acd91738cf3a1e212b844909c79bc86.

kuceb committed:
temp 02/07/19

kuceb committed:
add rightclick, fix trigger coords, add retry for flaky tests, add jquery fixes for focus

kuceb committed:
add rightclick, fix trigger coords, add retry for flaky tests, add jquery fixes for focus

kuceb committed:
add custom matchers helper, fix trigger, update test

kuceb committed:
remove only

kuceb committed:
add retries to flaky tests, remove debugging code

kuceb committed:
fix leaking defaultCommandTimeout

kuceb committed:
fix duplicate chai declaration

kuceb committed:
fix deps

kuceb committed:
Merge remote-tracking branch 'origin/issue-2957' into issue-311

kuceb committed:
base off simulated mouse fixes

kuceb committed:
fix type.js, fix deps

kuceb committed:
temp 04/18/19 [skip ci] click_spec passing

kuceb committed:
temp 04/22/19 [skip ci] actionability accept more options, designmode is focusable, clear modifiers

kuceb committed:
Add files via upload

kuceb committed:
temp 04/23/19 [skip ci] simulated default actions, change events correct for native event

kuceb committed:
Merge remote-tracking branch 'origin/issue-311' into issue-311

kuceb committed:
temp 04/24/19 [skip ci] simulated/native handle singlevaluechange inputs, reorganize

kuceb committed:
add webpack

kuceb committed:
fix build proccess

kuceb committed:
fix reporter build and test

kuceb committed:
fix reporter test .css

kuceb committed:
use apply(window, ...) since strict mode in webpack

kuceb committed:
temp 05/13/19 [skip ci] resolver

kuceb committed:
temp 05/14/19 [skip ci] webpack w/ root level symlinks

kuceb committed:
run ci

kuceb committed:
minor fixes

kuceb committed:
fix link-packages script

kuceb committed:
hopefully fix mac build

kuceb committed:
re-add build-prod script in reporter

kuceb committed:
fix typo in link-packages script

kuceb committed:
Merge branch 'develop' into webpack-runner

kuceb committed:
fix lint errors

kuceb committed:
fix node-version

kuceb committed:
namespace files in webpack

kuceb committed:
bust circle ci cache

kuceb committed:
bust circle ci cache 2

kuceb committed:
remove deps

kuceb committed:
bust circle ci cache 3

kuceb committed:
bust circle ci cache (launcher)

kuceb committed:
ok circleci, you win. I will change all the cache keys

kuceb committed:
update snapshot: not sure why added stack lines

kuceb committed:
add mocha to reporter devDeps

kuceb committed:
separate build file for reporter

kuceb committed:
remove unneeded code

kuceb committed:
node-sass-globbing use relative imports for appveyor

kuceb committed:
add debugging for appveyor failure

kuceb committed:
node-sass-globbing force unix paths for windows

kuceb committed:
node-sass-globbing force unix paths for windows 2

kuceb committed:
node-sass-globbing force unix paths for windows in runner

kuceb committed:
run ci

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
add checksum to packages caches

kuceb committed:
bust circle ci cache (sub packages)

kuceb committed:
Merge branch 'develop' into webpack-runner

kuceb committed:
testing code for appveyor env vars

kuceb committed:
Merge remote-tracking branch 'origin/develop' into fix-appveyor-build-script-only-on-develop

kuceb committed:
fix appveyor env vars for branchname
06-00000026

kuceb committed:
fix binary build script, smoke test catch errors, include pkg mains
03-0000000a

kuceb committed:
Merge branch 'fix-appveyor-build-script-only-on-develop' into webpack-runner

kuceb committed:
build binary in appveyor for this branch
07-00000047

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
fix binary unit test, revert mocha upgrade in root, fix e2e specs
02-00000004

kuceb committed:
statically replace symlinks at build time, add test, remove proxy packages code
03-00000002

kuceb committed:
oops, update snapshot
04-00000049

kuceb committed:
bump circle packages cache
02-00000005

kuceb committed:
fix path.win32 on windows, add test for it
00-00000058

kuceb committed:
Merge branch 'develop' into webpack-runner

kuceb committed:
for some reason packages ci cache is not including `rebuild-node-sass` in node_modules/.bin, and npm does not recreate .bin links if a node_modules folder already exists. so Im bumping all the npm caches, again
00-00000003

kuceb committed:
emit during webpack-dev-server watch mode
02-00000020

kuceb committed:
temp 05/22/19 [skip ci]

kuceb committed:
try revert circle.yml
02-0000002f

kuceb committed:
Merge branch 'develop' of github.com:cypress-io/cypress into webpack-runner

kuceb committed:
try revert everything
03-0000003b

kuceb committed:
Revert "try revert everything"

This reverts commit be125b88ff4b76e6d45ad5de4de82235f7816fb8.

kuceb committed:
no circle cache now
02-00000013

kuceb committed:
Merge branch 'webpack-runner' into issue-311

kuceb committed:
remove logging in reporter build
06-00002a41

kuceb committed:
Merge branch develop of github.com:cypress-io/cypress into webpack-runner
02-0000089b

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
make build scripts DRY
04-00000a7f

kuceb committed:
fix missing fonts, better watch mode
06-00003053

kuceb committed:
use webpack --progress in watch mode
00-00000582

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
move webpack config files into dedicated package
05-000023bf

kuceb committed:
build desktop-gui with webpack, use relative scss url()s
03-00001801

kuceb committed:
add timings info, fix configs
07-00005582

kuceb committed:
enable sourcemaps "eval", fix desktopgui test
04-00005a08

kuceb committed:
try 2: fix desktopgui test
01-0000058e

kuceb committed:
fix lint
03-00000b10

kuceb committed:
fix snapshot (very minor do not worry)
04-000014a4

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
code cleanup
00-00007a4f

kuceb committed:
try disable devtool
07-0000a97a

kuceb committed:
remove cypress:// devtool sourceURL mapping to prevent cross-domain on electron
00-000056e4

kuceb committed:
re-enable eval sourcemap in dev and prod
06-000051e6

kuceb committed:
code cleanup
02-000056ae

kuceb committed:
refactor node-jsdom setup for tests
00-000041a2

kuceb committed:
re-enable sourceURL protocol, enable crossorigin script on runner to appease electron

kuceb committed:
smoke tests for static assets in binary build script, disable sourceURL protocol mapping, more consistent dev/prod webpack builds

kuceb committed:
use cross-env for desktop-gui

kuceb committed:
desktop-dui: expect development NODE_ENV during build script (as has been in past)

kuceb committed:
fix build script unit test

kuceb committed:
fix build script unit test for real

kuceb committed:
fix sourcemaps for dev/prod, add comments, add build static asset tests

kuceb committed:
add cross-env to runner devDeps

kuceb committed:
fix unit tests for build binary script

kuceb committed:
fix whitespace

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
temp 06/07/19 [skip ci]

kuceb committed:
decaffeinate selection.js

kuceb committed:
support linting typescript

kuceb committed:
support linting typescript

kuceb committed:
temp 06/12/19 [skip ci] fix linting

kuceb committed:
use latest version of @typescript-eslint

kuceb committed:
better eslintrc, eslintignore

kuceb committed:
lint all the things

kuceb committed:
lint all the things(typescript)

kuceb committed:
fix lint script in ci

kuceb committed:
fix linting cli/types and coffeescript

kuceb committed:
remove subpackages "lint" scripts

kuceb committed:
remove un-needed dep

kuceb committed:
rename lint-typescript job to more accurate name

kuceb committed:
rename lint-typescript job to more accurate name pt 2

kuceb committed:
add postcss autoprefixer + staticAsset tets

kuceb committed:
add eslintrc to scripts/unit

kuceb committed:
tabs -> spaces

kuceb committed:
tabs -> spaces

kuceb committed:
tabs -> spaces

kuceb committed:
Merge branch 'lint-typescript' into issue-311

kuceb committed:
temp 06/13/19 [skip ci]

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
Merge branch 'lint-typescript' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner-lint-typescript' into issue-311

kuceb committed:
Revert "temp 06/13/19 [skip ci]"

This reverts commit 1a8316d0f0754c3400a4d8b82a199e73bec71f05.

kuceb committed:
temp 06/13/19 [skip ci] merge in typescript-lint, webpack runner, spec module

kuceb committed:
fix pre-commit lint hooks

kuceb committed:
fix pre-commit lint hooks

kuceb committed:
fix binary/util scripts

kuceb committed:
Merge branch 'webpack-runner' into issue-311

kuceb committed:
fix .prettierignore

kuceb committed:
Merge branch 'lint-typescript' into webpack-runner-lint-typescript

kuceb committed:
temp 06/14/19 [skip ci] spec modules

kuceb committed:
fix testStaticAssets in build script

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
Merge branch 'webpack-runner' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner-lint-typescript' into issue-311

kuceb committed:
fix extension requiring node socket

kuceb committed:
Merge branch 'webpack-runner' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner-lint-typescript' into issue-311

kuceb committed:
fix unit-test for binary build script

kuceb committed:
fix unit test for runner/reporter

kuceb committed:
Merge remote-tracking branch 'origin/develop' into lint-typescript

kuceb committed:
exclude transpiled .js files from eslint

kuceb committed:
Merge remote-tracking branch 'origin/develop' into lint-typescript

kuceb committed:
Merge remote-tracking branch 'origin/develop' into webpack-runner

kuceb committed:
Merge branch 'lint-typescript' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner-lint-typescript' into issue-311

kuceb committed:
remove only

kuceb committed:
temp 06/17/19 [skip ci] eslint changes

kuceb committed:
add rule: function-paren-newline:consistent

kuceb committed:
Merge branch 'lint-typescript' into issue-311

kuceb committed:
lint files

kuceb committed:
temp 06/17/19 [skip ci] foo

kuceb committed:
Merge remote-tracking branch 'origin/develop' into lint-typescript

kuceb committed:
temp 06/17/19 [skip ci]

kuceb committed:
remove un-needed eslintignore rules

kuceb committed:
Merge remote-tracking branch 'origin/develop' into lint-typescript

kuceb committed:
bump pkg version 'eslint-plugin-cypress-dev'

kuceb committed:
lint files

kuceb committed:
Merge remote-tracking branch 'origin/develop' into lint-typescript

kuceb committed:
Merge branch 'lint-typescript' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner' into webpack-runner-lint-typescript

kuceb committed:
Merge branch 'webpack-runner-lint-typescript' into issue-311

kuceb committed:
temp 06/18/19 [skip ci] resposition HEAD

kuceb committed:
foo only

kuceb committed:
Merge remote-tracking branch 'origin/develop' into issue-311

CypressJoseph commented:
@Bkucera any objections to closing this one?

Bkucera commented:
closing since this is very stale and no longer a blocker for cross-browser support (which is released)
